dotCMS/src/main/java/com/dotcms/xmlsitemap/XMLSitemapJob.java

```diff
@@ -238,11 +238,16 @@ public void generateSitemapPerHost() throws DotDataException, DotSecurityExcepti
 
 					//Getting the detail page, that detail page could be a HTMLPageAsset or a legacy page
 					IHTMLPage page = null;
+					List<Contentlet> results = APILocator.getContentletAPI()
+							.search("+identifier:" + st.getPagedetail() + " +live:true", 0, 0,
+									"moddate", systemUser, false);
+					if (results != null && !results.isEmpty()) {
 						//First lets asume it is a HTMLPageAsset
-						Contentlet contentlet = APILocator.getContentletAPI().search( "+identifier:" + st.getPagedetail() + " +live:true", 0, 0, "moddate", systemUser, false ).get( 0 );
-						if ( contentlet != null ) {
-							page = APILocator.getHTMLPageAssetAPI().fromContentlet( contentlet );
+						Contentlet contentlet = results.get(0);
+						if (contentlet != null) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)